src/tools/w3dviewer/w3dviewdoc.cpp

```diff
@@ -1018,23 +1021,181 @@ void CW3DViewDoc::SetFogColor(Vector3 &color)
     }
 }
 
-#if 0
-bool CW3DViewDoc::SaveAggregtate(const char *name)
+PrototypeClass *CW3DViewDoc::GenerateLOD(const char *name, bool format)
 {
-    // do later
-}
+    RenderObjIterator *iter = W3DAssetManager::Get_Instance()->Create_Render_Obj_Iterator();
 
-void CW3DViewDoc::GenerateLOD(const char *name, int type)
-{
-    // do later
+    if (iter == nullptr) {
+        return nullptr;
+    }
+
+    int count = 0;
+    int index = 0xFFFF;
+    char c = 'Z';
+
+    for (iter->First(); !iter->Is_Done(); iter->Next()) {
+        const char *robjname = iter->Current_Item_Name();
+
+        if (W3DAssetManager::Get_Instance()->Render_Obj_Exists(robjname)
+            && iter->Current_Item_Class_ID() == RenderObjClass::CLASSID_HLOD && strstr(robjname, name) == robjname) {
+            const char *str = &robjname[strlen(name)];
+
+            if (!format) {
+                if (format) {

```

Comment:
This looks suspicious. Essentially this condition will 100% be false because of the previous one, hence it would never be in a Release compiled binary.

Reply:
Ok now I see what's going on in CW3DViewDoc::GenerateLOD and I made it do what its clearly doing.